test/e2e/llmisvc/test_llm_inference_service.py

```diff
@@ -0,0 +1,297 @@
+
+
+def create_llmisvc(kserve_client, llm_isvc, namespace=None):
+    from kserve.utils import utils
+
+    version = llm_isvc.api_version.split("/")[1]
+
+    if namespace is None:
+        namespace = utils.get_isvc_namespace(llm_isvc)
+
+    try:
+        outputs = kserve_client.api_instance.create_namespaced_custom_object(
+            constants.KSERVE_GROUP,
+            version,
+            namespace,
+            KSERVE_PLURAL_LLMINFERENCESERVICE,
+            llm_isvc,
+        )
+        return outputs
+    except client.rest.ApiException as e:
+        raise RuntimeError(
+            f"Exception when calling CustomObjectsApi->"
+            f"create_namespaced_custom_object for LLMInferenceService: {e}"
+        ) from e
+
+
+def delete_llmisvc(
+    kserve_client, name, namespace, version=constants.KSERVE_V1ALPHA1_VERSION
+):
+    try:
+        return kserve_client.api_instance.delete_namespaced_custom_object(
+            constants.KSERVE_GROUP,
+            version,
+            namespace,
+            KSERVE_PLURAL_LLMINFERENCESERVICE,
+            name,
+        )
+    except client.rest.ApiException as e:
+        raise RuntimeError(
+            f"Exception when calling CustomObjectsApi->"
+            f"delete_namespaced_custom_object for LLMInferenceService: {e}"
+        ) from e
+
+
+def get_llmisvc(
+    kserve_client, name, namespace, version=constants.KSERVE_V1ALPHA1_VERSION
+):
+    try:
+        return kserve_client.api_instance.get_namespaced_custom_object(
+            constants.KSERVE_GROUP,
+            version,
+            namespace,
+            KSERVE_PLURAL_LLMINFERENCESERVICE,
+            name,
+        )
+    except client.rest.ApiException as e:
+        raise RuntimeError(
+            f"Exception when calling CustomObjectsApi->"
+            f"get_namespaced_custom_object for LLMInferenceService: {e}"
+        ) from e
+
+
+def wait_for_model_response(
+    kserve_client,
+    name,
+    namespace,
+    timeout_seconds=600,
+    version=constants.KSERVE_V1ALPHA1_VERSION,
+    model_name=None,
+):
+    if model_name is None:
+        model_name = "default-model"
+
+    service_url = None
+
+    def assert_model_responds():
+        nonlocal service_url
+
+        try:
+            service_url = get_llm_service_url(kserve_client, name, namespace, version)
+        except Exception as e:
+            raise AssertionError(f"Failed to get service URL: {e}") from e
+
+        completion_url = f"{service_url}/v1/completions"
+        test_payload = {"model": model_name, "prompt": "test", "max_tokens": 1}
+
+        try:
+            response = requests.post(
+                completion_url,
+                headers={"Content-Type": "application/json"},
+                json=test_payload,
+                timeout=30,
+            )
+        except Exception as e:
+            raise AssertionError(f"Failed to call model: {e}") from e
+
+        assert (
+            response.status_code == 200
```

Comment:
Another think that might be helpful is to plug compose-able response assertions

Reply:
Good idea. I'll make a struct for testing